packages/core/src/shader/ShaderData.ts

```diff
@@ -560,6 +561,20 @@ export class ShaderData implements IRefObject, IClone {
     }
   }
 
+  /**
+   * Get all shader properties that have been set on this shaderData
+   * @param out - All shader properties
+   */
+  getShaderProperties(out: ShaderProperty[]): void {
```

Comment:
You can return `out` for better code organization

Reply:
Maybe your actual needs is: 
```ts
const shaderProperties = getShaderProperties( [ ] );
```

I not very suggest to  return  `out param` in TS .   I write a method  overloaing, it's more easy to use:
```ts
const shaderProperties =  getShaderProperties();
```